src/util/initAlgoliaSearch.ts

```diff
@@ -0,0 +1,14 @@
+import AlgoliaClient from 'algoliasearch'
+import { Store } from 'redux'
+import { ALGOLIA_CONFIG } from '../constants'
+import { setRemoteSearch } from '../action-creators'
+import getSearchApiKey from '../search/getSearchApiKey'
+
+/**
+ * Initialize algolia search client.
+ */
+export const initAlgoliaSearch = async (userId: string, store: Store) => {
+  const apiKey = await getSearchApiKey(userId)
+  window.algoliaClient = AlgoliaClient(ALGOLIA_CONFIG.applicationId, apiKey)
```

Comment:
Let's store a singleton `algoliaClient` in a module which can be imported after it has been initialized (e.g. similar to `store.ts`). It's still global, but it's better than attaching to the `window` object.

Reply:
Sure. Will do that.